code/iaas/service-discovery/server/src/main/java/io/cattle/platform/servicediscovery/deployment/impl/DeploymentUnit.java

```diff
@@ -92,6 +95,34 @@ public boolean isError() {
         return false;
     }
 
+    public boolean isHostActive() {
+        for (DeploymentUnitInstance deployUnitInstance : getDeploymentUnitInstances()) {
+            if (deployUnitInstance instanceof ExternalDeploymentUnitInstance) {
```

Comment:
It should be if !(deploymentUnitInstance instanceof InstanceUnit). Otherwise the code will break when the new implementation of DeploymentUnitInstance that is not instance of InstanceUnit, gets added.